core/src/main/java/org/apache/accumulo/core/fate/zookeeper/MetaFateStore.java

```diff
@@ -335,18 +342,17 @@ public void setStatus(TStatus status) {
       verifyReservedAndNotDeleted(true);
 
       try {
-        zk.mutateExisting(getTXPath(fateId), currSerializedData -> {
-          NodeValue currNodeVal = new NodeValue(currSerializedData);
+        zk.mutateExisting(getTXPath(fateId), currSerFateData -> {
```

Comment:
Not sure if this would be effective, feel free to ignore this comment.  Was wondering if an internal util function like the following would reduce boiler place code a bit.  Can focus on transforming the FateData object.  Noticed a pattern that seems the same with this setStatus function, but not sure if its always the same.

```java
    private void mutate(FateId fateId, UnaryOperator<FateData<T>> fateDataOp) {
      try {
      zk.mutateExisting(getTXPath(fateId), currSerFateData -> {
        FateData<T> currFateData = new FateData<>(currSerFateData);
        FateData<T> newFateData = fateDataOp.apply(currFateData);
        if(newFateData == null){
          return null;
        }else{
          return newFateData.serialize();
        }

      });
      } catch (KeeperException | InterruptedException | AcceptableThriftTableOperationException e) {
        throw new IllegalStateException(e);
      }
    }
```

Reply:
Thanks, I added the suggested method. I think this cleans up the code and makes things a bit more readable and reusuable.